File: uf2/nf2/traduccio_bash/sumaParellSenar.java
import java.util.InputMismatchException;
import java.util.Scanner;

/**
 * Calcula (i mostra) la suma dels parells i senars en el rang entrat per l'usuari.
 * @author eina
 */

public class sumaParellSenar {
	public static void main(String[] args) {
		int parells = 0;
		int imparells = 0;
		int min = 0;
		int max = 0;
		boolean correcte = false;

		Scanner lector = new Scanner(System.in);
		try {
			System.out.print("Entra el valor mínim: ");
			min = lector.nextInt();

			System.out.print("Entra el valor màxim: ");
			max = lector.nextInt();

			if (min < max) {
				correcte = true;
			} else {
				System.out.println("El valor mínim ha de ser inferior al màxim.");
			}
		} catch (InputMismatchException e) {
			System.out.println("No s'ha pogut llegir el valor.");
		}

		if (correcte) {
			for (int i = min; i <= max; i++) {
				if (i %2 == 0) {
					parells += i;
				} else {
					imparells += i;
				}
			}
			System.out.printf("La suma dels nombres parells compresos entre %d i %d és %d i la dels imparells %d\n", min, max, parells, imparells);
		}
	}
}

File: uf2/nf2/traduccio_python/sumaParellSenar.py
#!/bin/python
# Rep com a arguments un valor mínim i un de màxim.
# Suma per separat els nombres parells i senars compresos en aquest rang i mostra el resultat.
# Retorna 1 si no se li passen dos arguments, 2 si no són enters i 3 si el segon no és major que el primer.
import sys

if len(sys.argv) != 3:
	print("Has d'introduir dos arguments.")
	sys.exit(1)

min = 0
max = 0

try:
	min = int(sys.argv[1])
	max = int(sys.argv[2])
except ValueError:
	print("Els arguments han de ser enters.")
	sys.exit(2)


if max < min:
	print("El segon argument ha de ser més gran que el primer.")
	sys.exit(3)

parells = 0
senars = 0

for i in range(min, max+1):
	if i % 2 == 0:
		parells += i
	else:
		senars += i

print("La suma del parells entre {} i {} és {} i la dels senars {}".format(min, max, parells, senars))
